### PR TITLE
BI-9193 Feature flags for LLP & LP journeys

### PR DESCRIPTION
**Pull request by phudson-ch**
- Fix routing for non authenticated user when LP & LLP feature flags are false

[BI-9193: feature flags for llp and lp journeys](https://companieshouse.atlassian.net/browse/BI-9193)

### CODE REVIEW

File: src/controllers/certificates/home.controller.ts [hunk 1]
```diff
@@ -60,12 +58,12 @@ export default async (req: Request, res: Response, next: NextFunction) => {
                 startNowUrl = replaceCompanyNumber(DISSOLVED_CERTIFICATE_TYPE, companyNumber);
                 logger.debug(`Certificates Home Controller - Dissolved Company, company_number=${companyNumber}, service_url=${SERVICE_URL}, start_now_url=${startNowUrl}`);
             } else {
-                if (DYNAMIC_LP_CERTIFICATE_ORDERS_ENABLED === "true" && companyType === "limited-partnership") {
+                if (FEATURE_FLAGS.lpCertificateOrdersEnabled && companyType === "limited-partnership") {
```
Comment (cgregoryCH) [inline src/controllers/certificates/home.controller.ts:61]: Could be refactored to utilise the strategy pattern

File: src/middleware/auth.middleware.ts [hunk 2]
```diff
@@ -24,9 +25,9 @@ export default async (req: Request, res: Response, next: NextFunction) => {
                 const companyProfile: CompanyProfile = await getCompanyProfile(API_KEY, companyNumber);
                 const companyType: string = companyProfile.type;
                 let returnToUrl: string;
-                if (CompanyType.LIMITED_PARTNERSHIP === companyType) {
+                if (featureFlags.lpCertificateOrdersEnabled && CompanyType.LIMITED_PARTNERSHIP === companyType) {
```
Comment (cgregoryCH) [inline src/middleware/auth.middleware.ts:28]: Can also use strategy pattern

**Review by cgregoryCH — COMMENTED** [review]
Consider suggested refactoring

File: src/controllers/certificates/home.controller.ts (hunk 1 above)
Comment (kinpang-CH) [reply src/controllers/certificates/home.controller.ts]: Resolved in latest commit.

File: src/middleware/auth.middleware.ts (hunk 2 above)
Comment (kinpang-CH) [reply src/middleware/auth.middleware.ts]: Resolved in latest commit.

**Review by kinpang-CH — APPROVED** [review]

**Review by cgregoryCH — APPROVED** [review]

**Pull request merged**